tests/phpunit/CRM/Core/BAO/MessageTemplateTest.php

```diff
@@ -397,6 +397,17 @@ public function hookTokenValues(array &$details): void {
     }
   }
 
+  /**
+   * Test that unresolved tokens are removed in the end.
```

Comment:
I should make it clear that I'm actually testing for lack-of-smarty fatal error here